### DISCUSSION TITLE
Kubeapps token getting stored locally

### DISCUSSION THREAD

**Discussion by athreyagithub** [kubeapps/kubeapps#4805] [Q&A] [answered]
I see when i login to kubeapps passing the token, the token is getting stored in the local storage in **kubeapps_auth_token**. how to avoid this? 
Also, kubeapps source codes are visible without any authentication. 

images are attached for the reference.
![kubeapps_token](https://user-images.githubusercontent.com/68467715/171162000-4bbcc5a8-1e75-4cd4-bc38-3d092d1a4099.PNG)
![kubeapps_source_code](https://user-images.githubusercontent.com/68467715/171162008-91013a5b-dbb1-4fbd-8679-f05339260070.PNG)

**Comment by antgamdia** [comment, marked as answer]
Hi @athreyagithub ,

the token is stored in the local storage, as you pointed out. This is the expected behavior: each time you call a Kubeapps API, this token will be used to authenticate each of these calls. What would you suggest?

Also, regarding the source code, it is not an issue at all: every single line of code is public in this repo, I don't think we should offuscate it, no? Or do you have any concerns? 

Anyway, please note that the token-based authentication is NOT intended for _production_ purposes. The recommended way is using OIDC to handle the authentication: https://github.com/vmware-tanzu/kubeapps/blob/main/site/content/docs/latest/tutorials/using-an-OIDC-provider.md

**Reply by athreyagithub** [reply]
Thanks for the quick reply. regarding source code, i understand, regarding the token getting stored i need to go with OIDC.
